src-electron/db/query-struct.js

```diff
@@ -154,6 +155,60 @@ async function selectStructByNameAndClusterId(db, name, clusterId, packageIds) {
   }
 }
 
+/**
+ * Select a struct matched by name and cluster name
+ * Note: Use selectStructByNameAndClusterName but this was needed for backwards compatibility.
+ * @param {*} db
+ * @param {*} name
+ * @param {*} clusterName
+ * @param {*} packageIds
+ * @returns struct information or undefined
+ */
+async function selectStructByNameAndClusterName(
+  db,
+  name,
+  clusterName,
+  packageIds
+) {
+  let queryWithClusterName = queryUtil.sqlQueryForDataTypeByNameAndClusterName(
+    dbEnum.zclType.struct,
+    name,
+    clusterName,
+    packageIds
+  )
+  let res = await dbApi
+    .dbAll(db, queryWithClusterName)
+    .then((rows) => rows.map(dbMapping.map.struct))
+  if (res && res.length == 1) {
+    return res[0]
+  } else {
+    queryWithClusterName = queryUtil.sqlQueryForDataTypeByNameAndClusterName(
+      dbEnum.zclType.struct,
+      name,
+      null, // Retrieving global data types since cluster specific ones were not found.
+      packageIds
+    )
+    res = await dbApi.dbGet(db, queryWithClusterName).then(dbMapping.map.struct)
+    if (!res) {
+      // Still no struct is found. Ignore where clause
+      // Note: Seems like there is a use case where the cluster xml file name
+      // is passed instead of cluster name in certain helpers arguments.
+      // Believe we should break here than actually satisfy this if condition.
```

Comment:
Ah, this is for the "backward compat names" bit.  This should be handled over in the Darwin bits, since it's specific to those.  I'll make a suggestion on that code.